pallets/liquid-staking/src/lib.rs

```diff
@@ -1342,40 +1552,37 @@ pub mod pallet {
             EraStartBlock::<T>::put(T::RelayChainValidationDataProvider::current_block_number());
             CurrentEra::<T>::mutate(|e| *e = e.saturating_add(offset));
 
-            let derivative_index = T::DerivativeIndex::get();
-            let unbonding_amount = Self::unbonding_of(derivative_index);
+            log::trace!(
+                target: "liquidStaking::do_advance_era",
+                "offset: {:?}",
+                &offset,
+            );
 
-            let (bond_amount, rebond_amount, unbond_amount) =
-                Self::matching_pool().matching(unbonding_amount)?;
-            if !StakingLedgers::<T>::contains_key(&derivative_index) {
-                Self::do_bond(derivative_index, bond_amount, RewardDestination::Staked)?;
-            } else {
-                Self::do_bond_extra(derivative_index, bond_amount)?;
-            }
+            Self::deposit_event(Event::<T>::NewEra(Self::current_era()));
 
-            Self::do_unbond(derivative_index, unbond_amount)?;
-            Self::do_rebond(derivative_index, rebond_amount)?;
+            // ignore error
+            let _ = || -> DispatchResult {
+                let unbonding_amount = Self::get_total_unbonding();
+                let (bond_amount, rebond_amount, unbond_amount) =
+                    Self::matching_pool().matching(unbonding_amount)?;
+                Self::do_multi_bond(bond_amount, RewardDestination::Staked)?;
+                Self::do_multi_rebond(rebond_amount)?;
 
-            Self::do_withdraw_unbonded(derivative_index, T::NumSlashingSpans::get())?;
+                Self::do_multi_unbond(unbond_amount)?;
 
-            Self::do_update_exchange_rate()?;
+                Self::do_multi_withdraw_unbonded(T::NumSlashingSpans::get())?;
 
```

Comment:
A concern here is, will this send too many xcm messages to relaychain?
if there are 8 accounts, the maximum number of xcm message could be 16,
including 8 unbond xcm and 8 withdraw_unbond xcm.
Also, we should benchmark the worst case for 'on_initial', and see how much weight does it cost.